### PR TITLE
[SPARK-29542][SQL][DOC] Make the descriptions of spark.sql.files.* be clearly

### PR DESCRIPTION
**Pull request by turboFei**
### What changes were proposed in this pull request?
As described  in [SPARK-29542](https://issues.apache.org/jira/browse/SPARK-29542) , the descriptions of `spark.sql.files.*` are confused.
In this PR, I make their descriptions be clearly.

### Why are the changes needed?
It makes the descriptions of `spark.sql.files.*` be clearly.
### Does this PR introduce any user-facing change?
No.

### How was this patch tested?
Existing UT.

### CODE REVIEW

**Comment by turboFei** [comment]
@wangyum

**Comment by turboFei** [comment]
Oh, In fact, all the parameters started with `spark.sql.file` are only effective on spark data source table.

Such as:

1. spark.sql.files.openCostInBytes
2. spark.sql.files.ignoreCorruptFiles
3. spark.sql.files.ignoreMissingFiles
4. spark.sql.files.maxRecordsPerFile

Is it conventional  that `files` represent `data source`?

**Comment by wangyum** [comment]
ok to test

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 1]
```diff
@@ -980,7 +980,8 @@ object SQLConf {
       .createWithDefault(true)
 
   val FILES_MAX_PARTITION_BYTES = buildConf("spark.sql.files.maxPartitionBytes")
-    .doc("The maximum number of bytes to pack into a single partition when reading files.")
+    .doc("The maximum number of bytes to pack into a single partition when reading files" +
+      " for data source table.")
```
Comment (wangyum) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: What if the built-in Parquet/ORC reader are used to process tables created by Hive(`spark.sql.hive.convertMetastoreParquet=true`)?

Comment (turboFei) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: Yes,  I have checked the configuration of the job I attached in JIRA.

We don't set spark.sql.hive.convertMetastoreParquet, so its value is true with default.

Comment (HyukjinKwon) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: I think @wangyum meant that, if we enable `spark.sql.hive.convertMetastoreParquet` or `spark.sql.hive.convertMetastoreOrc`, reading Hive table is also affected by this configurations. So, saying "for data source table" is not entirely correct.

"data source table" is also incorrect in the sense of DataFrame or Dataset.

Also, it does not support, IIRC, regular other data source tables. it only applies to file sources.

Comment (HyukjinKwon) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: I would simply describe this as, 

> ... partition when Spark file-based sources are used to read files

Comment (turboFei) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: thanks very much.

**Comment by HeartSaVioR** [comment]
Looks like the title needs to be updated to contain at least `[SQL]`. I'm not sure modifying description of config in code should be labeled as `[DOC]`.

**Comment by turboFei** [comment]
> Looks like the title of PR needs to be updated to contain at least `[SQL]`. I'm not sure modifying description of config in code should be labeled as `[DOC]` in the title of PR.

Thanks.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 2]
```diff
@@ -980,33 +980,36 @@ object SQLConf {
...
     .createWithDefault(128 * 1024 * 1024) // parquet.block.size
 
   val FILES_OPEN_COST_IN_BYTES = buildConf("spark.sql.files.openCostInBytes")
     .internal()
-    .doc("The estimated cost to open a file, measured by the number of bytes could be scanned in" +
-      " the same time. This is used when putting multiple files into a partition. It's better to" +
-      " over estimated, then the partitions with small files will be faster than partitions with" +
-      " bigger files (which is scheduled first).")
+    .doc("The estimated cost to open a file, measured by the number of bytes could be scanned in " +
+      "the same time. This is used when putting multiple file-source files into a partition. " +
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: file-source files sounds odd ...

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 3]
```diff
@@ -980,33 +980,36 @@ object SQLConf {
       .createWithDefault(true)
 
   val FILES_MAX_PARTITION_BYTES = buildConf("spark.sql.files.maxPartitionBytes")
-    .doc("The maximum number of bytes to pack into a single partition when reading files.")
+    .doc("The maximum number of bytes to pack into a single partition when Spark file-based " +
+      "sources are used to read files.")
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: Can you just revert everything here and add a sentence like "This configuration is effective only when using file-based sources such as Parquet, JSON and ORC." at the end of each configuration?

Comment (turboFei) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: make sense, thanks.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 4]
```diff
@@ -989,25 +991,31 @@ object SQLConf {
...
   val IGNORE_MISSING_FILES = buildConf("spark.sql.files.ignoreMissingFiles")
     .doc("Whether to ignore missing files. If true, the Spark jobs will continue to run when " +
-      "encountering missing files and the contents that have been read will still be returned.")
+      "encountering missing files and the contents that have been read will still be returned. " +
+      "This configuration is effective only when using file-based sources such as Parquet, JSON " +
+      "and ORC.")
     .booleanConf
     .createWithDefault(false)
 
   val MAX_RECORDS_PER_FILE = buildConf("spark.sql.files.maxRecordsPerFile")
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: I think this applies when we write Hive table as well. Can you double check?

**Review by HyukjinKwon — COMMENTED** [review]
Looks fine except https://github.com/apache/spark/pull/26200/files#r337833143

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala (hunk 4 above)
Comment (turboFei) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: sorry for mistake.
It applies for saveAsHiveFile too. And now there also some  some optimization that convert InsertIntoHiveTable to InsertFsRelationCommand. 
Thanks, I have revert the modification for this item.

Comment (turboFei) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: I also double check other items, they are ok.
Thanks a lot.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by wangyum** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
Merged to master and branch-2.4.

**Pull request closed without merging**